src/advanced_inv.cpp

```diff
@@ -1892,27 +1984,30 @@ bool advanced_inventory::query_charges( aim_location destarea, const advanced_in
             amount = std::min( weightmax, amount );
         }
     }
+    // handle how many of armour type we can equip (max of 2 per type)
+    if(destarea == AIM_WORN) {
+        const auto &id = sitem.items.front()->typeId();
+        amount = MAX_WORN_PER_TYPE - g->u.amount_worn(id);
```

Comment:
This causes a crash, it must be wrapped in a `std::min` call. The function must return not an amount that is larger than the available item count. If there is one item on the ground and it's wearable and you try to move it to the worn pane (with ENTER) or similar, this function is called and set `amount = 2` because `amount_worn` returns 0. That makes the caller move *2* items from the ground to the worn pane, which crashes when `remove_item` tries to remove two items from the ground, but there is only one.